src/server/pkg/collection/collection_test.go

```diff
@@ -27,8 +30,7 @@ var (
 )
 
 func TestIndex(t *testing.T) {
-	etcdClient, err := getEtcdClient()
-	require.NoError(t, err)
+	etcdClient := getEtcdClient()
```

Comment:
I started doing this to `getXClient()` methods in tests, after doing it in files that create a client once and then cache the error. When `getXClient()` simply caches an error and returns it in every subsequent invocation, there's no way to recover from the error, so returning it instead of panicking seemed pointless. I subsequently changed `pachyderm_test.go` and `driver/server_test.go`, because those tests never do anything with this error, and making it so that `getXClient()` doesn't return an error makes every test shorter (and means that you can do `getEtcdClient().DoEtcdOperation()` if you only need the client for one thing)

If you disagree, I'm not married to this opinion and I'm happy to change it back, certainly within this file

Reply:
Ah, I didn't realize we were caching the error. Makes sense